src/workers/helper.py

```diff
@@ -1,11 +1,27 @@
-from datetime import datetime
+from datetime import datetime, timezone, timedelta
+from dataclasses import dataclass
+import logging
+import asyncio
 import discord
+from discord.utils import snowflake_time
 from discord.ext.commands import Bot
 from settings import get_settings
 from util import create_interaction_embed_context
 from typing import Optional
 
+
+class UnableToDM(Exception):
+    pass
+
+
+@dataclass
+class AutomodResults:
+    num_removed: int = 0
+    num_errors: int = 0
```

Comment:
we could return num_removed and num_errors as a tuple in `automod_thread`

Reply:
Personally, I don't really think it's cleaner as a tuple. Here's a sample of what I had to change:
### `src/workers/forum_automod.py`
```python
    for channel_id, duration...:
        num_removed = 0
        num_errors = 0

        ...

        async for thread in...:
            num_removed, num_errors = await automod_thread(
                self, channel_id, thread, duration, num_removed, num_errors
            )
```

### `src/workers/helper.py`
```python
async def automod_thread(
    self,
    channel_id,
    thread: discord.Thread,
    duration: int,
    num_removed: int,
    num_errors: int,
):
    if thread.flags.pinned:
        return num_removed, num_errors

    ...

    if starter_message is not...:
        return num_removed, num_errors

    ....
    try:
        await thread.delete()
        ret = num_removed + 1, num_errors
        # code for dming user here
        return ret
    except...:
        return num_removed, num_errors + 1
```
If you think it's still better I'll add it though!

Reply:
imo i think it's fine because the benefit of having the class isn't really that big, but maybe @KingPieson has a differing opinion from me. while we're here we might as well type the return type in `automod_thread` function signature

Reply:
I'll make it a tuple then :nod: